src/allocator.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+This is a part of Loops project.
+Distributed under Apache 2 license.
+See https://github.com/vpisarev/loops/LICENSE
+*/
+
+#include "allocator.hpp"
+#include <string>
+#include <unistd.h>
+#include <sys/mman.h>
+#include <libkern/OSCacheControl.h>
+
+
+namespace loops
+{
+uint8_t* nonsys_alloc::allocate(size_t a_size)
+{
+    m_storage.push_back(std::vector<uint8_t>(a_size, 0));
+    std::vector<uint8_t>& holden = m_storage.back();
+    m_bufmap[&holden[0]] = &holden;
+    return &holden[0];
+}
+
+uint8_t* nonsys_alloc::expand(uint8_t* a_buffer, size_t a_size)
+{
+    if(m_bufmap.count(a_buffer) == 0)
+        throw std::string("Allocator: accessing non-existent buffer");
+    std::vector<uint8_t>& vec = *m_bufmap[a_buffer];
+    if(a_size == vec.size())
+        return a_buffer;
+    if(a_size < vec.size())
+        throw std::string("Allocator: buffer shrinking is not supported");
+    vec.resize(a_size, 0);
+    if(a_buffer != &vec[0])
+    {
+        m_bufmap.erase(a_buffer);
+        a_buffer = &vec[0];
+        m_bufmap[a_buffer] = &vec;
+    }
+    return a_buffer;
+}
+
+void nonsys_alloc::erase(uint8_t* a_buffer)
```

Comment:
erase is not needed in loops, at least in the short-term. We will only be adding new code, not removing any old code. Please, take it into account and make the design simpler and more robust

Reply:
Done.